subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/DefaultLenientConfiguration.java

```diff
@@ -373,7 +373,15 @@ public static class ArtifactResolveException extends ResolveException {
         private final String type;
         private final String displayName;
 
-        public ArtifactResolveException(String type, String path, String displayName, Iterable<Throwable> failures) {
+        public ArtifactResolveException(String type, String displayName, Iterable<? extends Throwable> failures) {
+            super(displayName, failures);
+            this.type = type;
+            this.displayName = displayName;
+        }
+
+        // This constructor can be removed.
```

Comment:
❓ Is there some reason it should not be removed?

Reply:
Wanted to avoid unnecessary changes, since this was originally part of a larger set of changes and I wanted to keep the changeset small. 

However since this is pulled out now, I will remove usages of this constructor